CliWrap.Tests/ExitConditionSpecs.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Xunit;
+
+namespace CliWrap.Tests;
+
+public class ExitConditionSpecs()
+{
+    [Fact(Timeout = 15000)]
+    public async Task I_can_execute_a_command_that_creates_child_process_resuing_standard_output_and_finish_after_child_process_exits()
+    {
+        // Arrange
+        var cmd = this.PrepareCommand(line => { });
+
+        // Act
+        var executionStart = DateTime.UtcNow;
+        var result = await cmd.ExecuteAsync();
```

Comment:
`executionStart`/`executionFinish` is already provided by `result`

Reply:
Yes, but finish time is a different one. CliWrap sets `ExitTime` when the main process ends. When there are child processes attached to output, `ExecuteAsync()` is blocked until all children are completed. In this test, we want to measure time when the `ExecuteAsync()` was blocked for, so that we can verify whether it was waiting for child or not.

Reply:
Exit time is set when the child process exits: https://github.com/Tyrrrz/CliWrap/blob/1cfa50eaaf8b8c74522a999e89dcf3577602eae3/CliWrap/Utils/ProcessEx.cs#L47-L51

It shouldn't be affected by the exit condition you've added, since that only affects whether CliWrap waits for pipes to clear or not.

Because your `var executionFinish = ...` statement appears right `await cmd.ExecuteAsync()`, it's essentially the same (slightly later) timestamp as the one provided by `ExecutionResult`.

Reply:
CliWrap's ExitTime is always the same no matter what `ExitCondition` is set. So this test would basically always pass.

Reply:
I'm confused in that case. Wouldn't `ExitCondition.ProcessExit` make the command finish _earlier_, assuming it spawns a grandchild process?